exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/ProofMapIndexProxyNoKeyHashingIntegrationTest.java

```diff
@@ -0,0 +1,482 @@
+      List<MapEntry<HashCode, String>> entries = createMapEntries(
+          Stream.of(
+              proofKeyFromPrefix("0000|01"),
+              proofKeyFromPrefix("0000|11"),
+              proofKeyFromPrefix("0010|00"),
+              proofKeyFromPrefix("0010|10")
+          )
+      );
+
+      putAll(map, entries);
+
+      List<HashCode> proofKeys = Arrays.asList(
+          // Should be rejected on root level
+          proofKeyFromPrefix("01|0000"),
+          // Should be rejected on intermediate level
+          proofKeyFromPrefix("00|01"),
+          proofKeyFromPrefix("00|11"),
+          // Should be rejected on leaf level
+          proofKeyFromPrefix("0000|00"),
+          proofKeyFromPrefix("0000|10"),
+          proofKeyFromPrefix("0010|01"),
+          proofKeyFromPrefix("0010|11")
+      );
+
+      assertThat(map, provesThatAbsent(proofKeys));
+    });
+  }
+
+  @CiOnly
+  @Test
+  /*
+    Takes quite a lot of time (validating 257 proofs), but it's an integration test, isn't it? :-)
+    Consider adding a similar test for left-leaning MPT
+   */
+  void getProof_MapContainsRightLeaningMaxHeightMpt() {
+    runTestWithView(database::createFork, (map) -> {
+      List<MapEntry<HashCode, String>> entries = createEntriesForRightLeaningMpt();
+      putAll(map, entries);
+
+      for (MapEntry<HashCode, String> e : entries) {
+        assertThat(map, provesThatPresent(e.getKey(), e.getValue()));
+      }
+    });
+  }
+
+  /**
+   * Returns a new key with the given prefix.
+   *
+   * @param prefix a key prefix — from the least significant bit to the most significant,
+   *        i.e., "00 01" is 8, "10 00" is 1.
+   *        May contain spaces, underscores or bars (e.g., "00 01|01 11" and "11_10"
+   *        are valid strings).
+   */
+  private static HashCode proofKeyFromPrefix(String prefix) {
+    prefix = filterBitPrefix(prefix);
+    byte[] key = keyFromString(prefix);
+    return HashCode.fromBytes(key);
+  }
+
+  /**
+   * Replaces spaces that may be used to separate groups of binary digits.
+   */
+  private static String filterBitPrefix(String prefix) {
+    String filtered = prefix.replaceAll("[ _|]", "");
+    // Check that the string is correct
+    assert filtered.matches("[01]*");
+    assert filtered.length() <= PROOF_MAP_KEY_SIZE;
+    return filtered;
+  }
+
+  /**
+   * Creates a 32-byte key from the bit prefix.
+   */
+  private static byte[] keyFromString(String prefix) {
+    BitSet keyPrefixBits = new BitSet(prefix.length());
+    for (int i = 0; i < prefix.length(); i++) {
+      char bit = prefix.charAt(i);
+      if (bit == '1') {
+        keyPrefixBits.set(i);
+      }
+    }
+    return createPrefixed(keyPrefixBits.toByteArray(), PROOF_MAP_KEY_SIZE);
+  }
+
+  /**
+   * Creates `numOfEntries` map entries, sorted by key:
+   * [(00…0PK1, V1), (00…0PK2, V2), … (00…0PKi, Vi)].
+   */
+  List<MapEntry<HashCode, String>> createSortedMapEntries() {
+    // Use PROOF_KEYS which are already sorted.
```

Comment:
Use them directly instead of interface method, with appropriate
    naming there won't be a need for comment